garrysmod/gamemodes/base/entities/entities/base_trigger.lua

```diff
@@ -0,0 +1,95 @@
+
+ENT.Base = "base_entity"
+ENT.Type = "trigger"
+
+--[[---------------------------------------------------------
+	Name: Initialize
+-----------------------------------------------------------]]
+function ENT:Initialize()
+	self:SetTrigger(true)
+	self.Entities = {}
+end
+
+--[[---------------------------------------------------------
+	Name: IsTouchedBy
+-----------------------------------------------------------]]
+function ENT:IsTouchedBy(ent)
+	return self.Entities[ent]
+end
+
+--[[---------------------------------------------------------
+	Name: StartTouch
+	Desc: Called when a entity starts touching us
+-----------------------------------------------------------]]
+function ENT:StartTouch(ent)
+	if ( !self:PassesTriggerFilters(ent) ) then return end
+	self.Entities[ent] = ent
+	
+	self:Input("OnStartTouch", self, ent)
+end
+
+--[[---------------------------------------------------------
+	Name: EndTouch
+	Desc: Called when a entity stops touching us
+-----------------------------------------------------------]]
+function ENT:EndTouch(ent)
+	if ( !self:IsTouchedBy(ent) ) then return end
+	self.Entities[ent] = nil
+	
+	self:Input("OnEndTouch", self, ent)
+
+	local bFoundOtherTouchee = false
+	local iSize = #self.Entities
+	for i=iSize, 0, -1 do 
+		local hOther = self.Entities[i]
+		if ( !IsValid(hOther) ) then
+			self.Entities[hOther] = nil
+		else
+			bFoundOtherTouchee = true
+		end
+	end
+
+	if ( !bFoundOtherTouchee ) then
+		self:Input("OnEndTouchAll", self, ent)
+	end
+end
+
+--[[---------------------------------------------------------
+	Name: Touch
+	Desc: Called when a entity touches us
+-----------------------------------------------------------]]
+function ENT:Touch(ent)
+	if ( !self:PassesTriggerFilters(ent) ) then return end
+	if ( !self.Entities[ent] ) then self.Entities[ent] = ent end
```

Comment:
He could, but doing this saves the overhead of that function call. It might make sense if we want more logic for `IsTouchedBy`, but for now it's probably worth it to duplicate this code.